### PR TITLE
VZ-3241.  Move COH operator install to platform operator

### PR DESCRIPTION
**Pull request by michael-cico**
# Description

Fixes VZ-3240.

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/scripts/install/3-install-verrazzano.sh [hunk 1]
```diff
@@ -312,7 +312,7 @@ if [ "${REGISTRY_SECRET_EXISTS}" == "TRUE" ]; then
 fi
 
 action "Installing Verrazzano system components" install_verrazzano || exit 1
-action "Installing Coherence Kubernetes operator" install_coherence_operator || exit 1
+platform_operator_install_message "Coherence Kubernetes operator"
```
Comment (bobdonat) [inline platform-operator/scripts/install/3-install-verrazzano.sh:315]: Does the new platform-operator-install handle slow image pulls?  The scripts were checking for slow image pulls and retrying the helm install if needed.

File: platform-operator/controllers/verrazzano/component/registry.go [hunk 2]
```diff
@@ -98,6 +98,9 @@ func GetComponents() []Component {
 			chartDir:                filepath.Join(thirdPartyChartsDir, "coherence-operator"),
 			chartNamespace:          constants.VerrazzanoSystemNamespace,
 			ignoreNamespaceOverride: true,
+			supportsOperatorInstall: true,
+			waitForInstall:          true,
+			imagePullSecretKeyname:  "imagePullSecrets[0].name",
```
Comment (bobdonat) [inline platform-operator/controllers/verrazzano/component/registry.go:103]: does this assume no imagePullSecrets are specified when installing the Coherence operator

File: platform-operator/scripts/install/3-install-verrazzano.sh (hunk 1 above)
Comment (michael-cico) [reply platform-operator/scripts/install/3-install-verrazzano.sh:315]: Probably not, we'll need to account for that.  Right now we're just starting to transition things.

File: platform-operator/controllers/verrazzano/component/helm_component.go [hunk 3]
```diff
@@ -190,7 +190,7 @@ func (h helmComponent) Upgrade(log *zap.SugaredLogger, client clipkg.Client, ns
 	log.Infof("Created values file: %s", tmpFile.Name())
 
 	// Perform a helm upgrade --install
-	_, _, err = upgradeFunc(log, h.releaseName, namespace, h.chartDir, h.waitForInstall, dryRun, overridesString, h.valuesFile, tmpFile.Name())
+	_, _, err = upgradeFunc(log, h.releaseName, namespace, h.chartDir, true, dryRun, overridesString, h.valuesFile, tmpFile.Name())
```
[the remark this answers is not in the document]
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/helm_component.go:193]: Yes, to replicate what we're doing in the scripts now.  This is likely to change later on during the 1.1 release.

File: platform-operator/controllers/verrazzano/component/registry.go (hunk 2 above)
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/registry.go:103]: The logic for image pull secrets is the same in the operator.  If the global pull secret is detected, we add the pull secret helm override to the install.

**Review by pfmackin — APPROVED** [review]
Looks ok, please track the slow image pull work

File: platform-operator/scripts/install/3-install-verrazzano.sh (hunk 1 above)
Comment (michael-cico) [reply platform-operator/scripts/install/3-install-verrazzano.sh:315]: Opened VZ-3485 to track this during the migration.

**Pull request merged**